dialect/sql/scan.go

```diff
@@ -29,6 +30,10 @@ func ScanOne(rows ColumnScanner, v interface{}) error {
 		return fmt.Errorf("sql/scan: unexpected number of columns: %d", n)
 	}
 	if !rows.Next() {
+		if rows.Err() != nil {
```

Comment:
This is an extremely odd case. The error on rows is only present after the call to `Next`